test-tools/wamr-ide/VSCode-Extension/src/debugConfigurationProvider.ts

```diff
@@ -9,50 +9,34 @@ import * as os from 'os';
 export class WasmDebugConfigurationProvider
     implements vscode.DebugConfigurationProvider
 {
-    /* default port set as 1234 */
-    private port = 1234;
-    private hostPath!: string;
-    private providerPromise: Thenable<vscode.DebugConfiguration> | undefined =
-        undefined;
-
     private wasmDebugConfig!: vscode.DebugConfiguration;
 
-    public resolveDebugConfiguration():
-        | Thenable<vscode.DebugConfiguration>
-        | undefined {
-        if (!this.providerPromise) {
-            this.providerPromise = Promise.resolve(this.wasmDebugConfig);
-            return this.providerPromise;
-        }
-        return this.providerPromise;
-    }
+    resolveDebugConfiguration(
+        _: vscode.WorkspaceFolder | undefined,
+        debugConfiguration: vscode.DebugConfiguration,
+    ): vscode.ProviderResult<vscode.DebugConfiguration> {
+        const defaultConfig: vscode.DebugConfiguration = {
+            type: 'wamr-debug',
+            name: 'Attach',
+            request: 'attach',
+            stopOnEntry: true,
+            attachCommands: [
+                /* default port 1234 */
+                'process connect -p wasm connect://127.0.0.1:1234',
+            ]
+        };
 
-    public setDebugConfig(hostPath: string, port: number): void {
-        this.port = port;
-        this.hostPath = hostPath;
         /* linux and windows has different debug configuration */
         if (os.platform() === 'win32' || os.platform() === 'darwin') {
```

Comment:
The `os.platform() === 'linux'` branch has been removed, does it work on linux without that?

Reply:
`['initCommands']: ['platform select remote-linux']` was the only attributes that was added to the configuration-object in that branch. The rest was redundant. 
So if if `os.platform() === 'win32' || os.platform() === 'darwin'` is true, we add the respective additional configuration to the default.